drivers/comparator/comparator_stm32_comp.c

```diff
@@ -0,0 +1,376 @@
+#endif
+
+#define STM32_COMP_DT_INST_P_IN(inst)                                                              \
+	CONCAT(LL_COMP_INPUT_PLUS_, DT_INST_STRING_TOKEN(inst, positive_input))
+
+#define STM32_COMP_DT_INST_N_IN(inst)                                                              \
+	CONCAT(LL_COMP_INPUT_MINUS_, DT_INST_STRING_TOKEN(inst, negative_input))
+
+#define STM32_COMP_DT_INST_HYST_MODE(inst)                                                         \
+	CONCAT(LL_COMP_HYSTERESIS_, DT_INST_STRING_TOKEN(inst, hysteresis))
+
+#define STM32_COMP_DT_INST_INV_OUT(inst)                                                           \
+	CONCAT(LL_COMP_OUTPUTPOL_, DT_INST_STRING_TOKEN(inst, invert_output))
+
+#define STM32_COMP_DT_INST_BLANK_SEL(inst)                                                         \
+	CONCAT(LL_COMP_BLANKINGSRC_, DT_INST_STRING_TOKEN(inst, st_blank_sel))
+
+#define STM32_COMP_DT_INST_LOCK(inst) DT_INST_PROP(inst, st_lock_enable)
+
+#define STM32_COMP_DT_MILLER_EFFECT_HOLD_ENABLE(inst)                                              \
+	DT_INST_PROP(inst, st_miller_effect_hold_enable)
+
+#define STM32_COMP_DT_EXTI_LINE(inst) CONCAT(LL_EXTI_LINE_, DT_INST_PROP(inst, st_exti_line))
+
+#define STM32_COMP_DT_EXTI_LINE_NUMBER(inst) DT_INST_PROP(inst, st_exti_line)
+
+#define STM32_COMP_DT_POWER_MODE(inst)                                                             \
+	CONCAT(LL_COMP_POWERMODE_, DT_INST_STRING_TOKEN(inst, st_power_mode))
+
+#if defined(SOC_SERIES_STM32H7X)
+#define STM32_COMP_DT_INST_COMP_CONFIG_INIT(inst)                                                  \
+	{                                                                                          \
+		.PowerMode = STM32_COMP_DT_POWER_MODE(inst),                                       \
+		.InputPlus = STM32_COMP_DT_INST_P_IN(inst),                                        \
+		.InputMinus = STM32_COMP_DT_INST_N_IN(inst),                                       \
+		.InputHysteresis = STM32_COMP_DT_INST_HYST_MODE(inst),                             \
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#else /* SOC_SERIES_STM32H7X */
+#define STM32_COMP_DT_INST_COMP_CONFIG_INIT(inst)                                                  \
+	{                                                                                          \
+		.InputPlus = STM32_COMP_DT_INST_P_IN(inst),                                        \
+		.InputMinus = STM32_COMP_DT_INST_N_IN(inst),                                       \
+		.InputHysteresis = STM32_COMP_DT_INST_HYST_MODE(inst),                             \
+		.OutputPolarity = STM32_COMP_DT_INST_INV_OUT(inst),                                \
+		.OutputBlankingSource = STM32_COMP_DT_INST_BLANK_SEL(inst),                        \
+	}
+#endif /* SOC_SERIES_STM32H7X */
```

Comment:
If the SoC specific handling is caused by the h7 comp IP being different from other STM32, please introduce a `st,stm32h7-comp` binding that can be used to distinguish implementation to use.
The interest is that for the next SoC series that has an IP similar to H7, we just have to use `st,stm32h7-comp` compatible in dts description and "voilà": no impact required in the driver to support the new series.

Reply:
Added `st,stm32h7-comp` and corresponding dt combat preprocessor check.